Develop

cristiancastano852 opened the pull request:
Etapa 2: ACLs y pagos simples

En esta fase, se añadirá el control de acceso (ACL) para gestionar los permisos de los roles y se implementarán las funcionalidades de pago simple. Los administradores podrán definir qué información se requiere de los usuarios para realizar pagos simples y los usuarios podrán efectuar estos pagos.

Funcionalidades del administrador
Definir los campos solicitados a los usuarios al momento de realizar un pago simple en un sitio abierto.
- Información del pagador: nombre, tipo y número de documento, correo.
- Configuración de pagos: moneda, tiempo de expiración del pago.
Ver la lista de transacciones realizadas en los diferentes micrositios.
Funcionalidades del usuario (invitado)
Diligenciar el formulario y generar una sesión de pago básica.
Validar los datos de entrada según las reglas del administrador.
Ver la información de su transacción en una página de resumen.

cristiancastano852 committed:
feat: implement connection with placetopay

cristiancastano852 committed:
feat: implement payment show

cristiancastano852 committed:
fix: problem with ci import

cristiancastano852 committed:
feat: improve the code

cristiancastano852 committed:
feat: decrease version to php

cristiancastano852 committed:
feat_ refresh cache in github actions

cristiancastano852 committed:
feat: implement get method in PaymentGateway

cristiancastano852 committed:
fix: step list files in gateway

cristiancastano852 committed:
feat: rename PlacetopayGateway.php to PlaceTopayGateway.php

cristiancastano852 committed:
fix: rename PlaceTopayGateway.php to PlacetoPayGateway.php

cristiancastano852 committed:
feat: add ExceptionHandling to test

cristiancastano852 committed:
feat: improve test with mocks

cristiancastano852 committed:
feat: update composer

cristiancastano852 committed:
feat: update with phpunit

cristiancastano852 committed:
fix: update name in file test

cristiancastano852 committed:
feat: update test with factory

cristiancastano852 committed:
feat: update sonar cloud properties

cristiancastano852 committed:
feat: ignore code

cristiancastano852 committed:
feat: add new field to microsites migration

cristiancastano852 committed:
feat: config spartan in vue

cristiancastano852 committed:
feat: add new view with vue.js and spartan

cristiancastano852 committed:
feat: change role views to vue

cristiancastano852 committed:
feat: improve code with pin

cristiancastano852 committed:
feat: create microsite with fields config

cristiancastano852 committed:
fix: make sure that using this pseudorandom

cristiancastano852 committed:
fix: encode trankey to paypal

cristiancastano852 committed:
Merge pull request #19 from cristiancastano852/feature/implement-fields-to-forms

Feature/implement fields to forms

cristiancastano852 committed:
Merge pull request #18 from cristiancastano852/feature/connect_to_external_payment-gateways

Feature/connect to external payment gateways

cristiancastano852 committed:
feat: create form to pay with dinamic fields

cristiancastano852 committed:
fix: update test with payment

cristiancastano852 committed:
fix: update test

cristiancastano852 committed:
feat: get all transactions

cristiancastano852 committed:
feat: add guest role to new registered users

cristiancastano852 committed:
Merge pull request #20 from cristiancastano852/feature/create-form-to-payment

Feature/create form to payment

cristiancastano852 committed:
feat: config permissions

cristiancastano852 committed:
feat: user laravel permissions

cristiancastano852 committed:
Merge branch 'develop' into feature/add-laravel-permissions-to-vue

cristiancastano852 committed:
feat: Add date-fns dependency for date manipulation

cristiancastano852 committed:
Merge pull request #21 from cristiancastano852/feature/add-laravel-permissions-to-vue

Feature/add laravel permissions to vue

cristiancastano852 committed:
feat: add logs in placetopay

cristiancastano852 committed:
feat: improve code with bin and test

cristiancastano852 committed:
feat: Add language translation

cristiancastano852 committed:
feat: Refactor PaymentGateway and QueryPaymentResponse toArray methods

cristiancastano852 committed:
Merge pull request #22 from cristiancastano852/feature/improve-payment-with-logs

Feature/improve payment with logs

barbosa89 commented on app/Http/Controllers/PaymentController.php:
```diff
@@ -0,0 +1,86 @@
+        ]);
+    }
+
+    public function transactionsByMicrosite($microsite_id): \Inertia\Response
```
Mover a otro controlador, puede ser MicrositePaymentController

barbosa89 commented on app/Http/Requests/Auth/LoginRequest.php:
```diff
@@ -63,16 +63,16 @@ public function ensureIsNotRateLimited(): void
-        ]);
+        // throw ValidationException::withMessages([
+        //     'email' => trans('auth.throttle', [
+        //         'seconds' => $seconds,
```
Limpiar código

barbosa89 commented on app/Http/Requests/StorePaymentRequest.php:
```diff
@@ -0,0 +1,38 @@
+    {
+        return [
+            // 'description' => ['required', 'string', 'min:3', 'max:100'],
+            // 'amount' => ['required', 'integer', 'min:1', 'max:999999999999'],
```
Si no tienes datos por validar, lo puedes eliminar

barbosa89 commented on app/Services/Payments/Gateways/PaypalGateway.php:
```diff
@@ -0,0 +1,98 @@
+    {
+        // $url = $this->config['url'].'/'.$payment->process_identifier;
+
+        // $response = Http::post($url, $this->data);
```
Limpiar código

barbosa89 commented on tests/Feature/Payments/StorePaymentTest.php:
```diff
@@ -0,0 +1,314 @@
+        ]);
+    }
+
+    public function testItHandlesClientError(): void
```
Si el servidor responde con 500, el error no fue manejado

cristiancastano852 committed:
feat: add MicrositePaymentController for handling transactions by microsite

- Added a new file `MicrositePaymentController.php` in the `app/Http/Controllers` directory.
- The `MicrositePaymentController` class handles the retrieval of transactions by microsite.
- It retrieves payments associated with a specific microsite ID and includes the microsite information.
- The controller uses the `Payment` and `Microsites` models.
- The `transactionsByMicrosite` method returns an Inertia response with the payments and microsites data.

Refactor PaymentController to remove transactionsByMicrosite method

- Removed the `transactionsByMicrosite` method from the `PaymentController` class.
- This method is now handled by the newly added `MicrositePaymentController`.

Remove unnecessary code in LoginRequest

- Removed unused imports and commented out code in the `LoginRequest` class.

Remove unnecessary validation rules in StorePaymentRequest

- Removed commented out validation rules in the `StorePaymentRequest` class.

Update PlacetoPayGateway to use new API endpoint

- Updated the `PlacetoPayGateway` class to use a new API endpoint for processing payments.
- The `process` method now sends a POST request to the updated endpoint.

Update PaymentService to remove unnecessary code

- Removed unnecessary code in the `PaymentService` class.
- The `create` method no longer includes an empty process.

Update routes to use MicrositePaymentController

- Updated the route for handling transactions by microsite to use the `MicrositePaymentController` instead of the `PaymentController`.

Update StorePaymentTest to assert correct status and errors

- Updated the `testItHandlesClientError` and `testItHandlesUnknownError` methods in the `StorePaymentTest` class.
- The tests now assert the correct status code and session errors.

cristiancastano852 committed:
refactor: improve code with bin

cristiancastano852 committed:
Merge pull request #27 from cristiancastano852/refactor/tests-and-methods-develop-to-master-release-v2.0.0

Resolve comments in PR develop to master release v2
